### DISCUSSION TITLE
Geographic things

### DISCUSSION THREAD

**Discussion by cbertelli** [simonw/datasette#951] [General]
Hello,
I think it could be interesting to take a walk on [gaia-gis.it](https://www.gaia-gis.it/). Spatialite is a very interesting project, it has been just updated to version 5 and could provide some useful conversion tools in addition to what you are already using (eg. virtual_pg).
By the way [gisquick.org](http://gisquick.org/) automates publishing of SQLite/Spatialite datasets.
I do appreciate very much the approach and results of Datasette. Thank you very much for providing it.
I hope someone does what you did for SQLite even for PostgreSQL.

**Comment by simonw** [comment]
Hi @cbertelli,

I've not tried it with version 5 yet, but I've done a bunch of work with it using SpatiaLite 4.3. There's some documentation about using Datasette with SpatiaLite here: https://docs.datasette.io/en/stable/spatialite.html

I actually spent some time this weekend on a Datasette+SpatiaLite project: https://calands.datasettes.com/ - it's a Datasette instance that exposes the California Protected Areas Database from https://www.calands.org/cpad/

That database is published as a shapefile, so I used my [shapefile-to-sqlite](https://pypi.org/project/shapefile-to-sqlite/) tool to convert that to SpatiaLite.

It's also using the [datasette-leaflet-geojson](https://github.com/simonw/datasette-leaflet-geojson) plugin to render maps, e.g. on https://calands.datasettes.com/calands/superunits_with_maps?_search=golden+gate

<img width="1155" alt="calands__superunits_with_maps__4_rows_where_where_search_matches__golden_gate_" src="https://user-images.githubusercontent.com/9599/91484194-bab13680-e85d-11ea-871f-18899eee49fc.png">

I'm really interested in investing more time in SpatiaLite stuff - I'd love to make it easier to build things like "search for geometries near this point" or "search for geometries within this bounding box". I'm trying to work out how to do "parks near me" using a spatial index in this ticket: https://github.com/simonw/calands-datasette/issues/3

**Comment by simonw** [comment]
I wrote about that project in my weeknotes today: https://simonwillison.net/2020/Aug/28/weeknotes-cpad/

**Comment by cbertelli** [comment]
Yes, it's the approach that is truly remarkable. Thanks for sharing.
c

**Comment by cbertelli** [comment]
By the way, I see you have an answer also for my other wish/question, a
Datasette for PostgreSQL here:
https://github.com/simonw/datasette/issues/670
It seems to me a similar questions to those we discussed with Marcello
Verona about the heuristics of vFront (https://github.com/marciuz/vfront).
It's an old PHP project and it doesn't look at the introspection features
of version 9.6 onwards, but it could be useful.
Anyway, I would really like very much your approach applied to PostgreSQL,
the use case is research data inside a wiki. I'm looking to Documize, but
the question is more general, getting data inside documents; the idea is
embedding like Medium does, but even there, databases are not welcome.
c

**Reply by simonw** [reply]
Have you tried using db-to-sqlite to export your PostgreSQL database for use with Datasette?

I use it for my blog (around 20,000 rows) and it only takes a few seconds - hast enough that I can run it every few hours to keep a copy around that I can use with Datasette.

https://github.com/simonw/db-to-sqlite
